pkg/master/controller.go

```diff
@@ -117,6 +117,9 @@ func (c *Controller) UpdateKubernetesService() error {
 		if err := c.CreateMasterServiceIfNeeded("kubernetes", c.ServiceIP, servicePorts, serviceType); err != nil {
 			return err
 		}
+		if err := c.SetService("kubernetes", servicePorts, serviceType); err != nil {
```

Comment:
how will this work if multiple masters have differing "extra ports"? how would you go about updating that setting if you needed to? You couldn't restart all the masters simultaneously, and you wouldn't want outdated masters stomping the new setting